packages/clerk-js/src/ui/components/PaymentSources/PaymentSources.tsx

```diff
@@ -0,0 +1,198 @@
+  const { data: paymentSources } = data || { data: [] };
+
+  return (
+    <ProfileSection.Root
+      title={localizationKeys('userProfile.__experimental_billingPage.paymentSourcesSection.title')}
+      centered={false}
+      id='profile'
+      sx={{ paddingTop: 0, borderTop: 'none', flex: 1 }}
+    >
+      <Action.Root>
+        <ProfileSection.ItemList id='paymentSources'>
+          {isLoading ? (
+            <Flex
+              direction={'row'}
+              align={'center'}
+              justify={'center'}
+              sx={t => ({
+                width: '100%',
+                minHeight: t.sizes.$60,
+              })}
+            >
+              <Spinner
+                size={'lg'}
+                colorScheme={'primary'}
+                elementDescriptor={descriptors.spinner}
+              />
+            </Flex>
+          ) : (
+            <>
+              {paymentSources.map(paymentSource => (
+                <Fragment key={paymentSource.id}>
+                  <ProfileSection.Item id='paymentSources'>
+                    <Flex
+                      sx={{ overflow: 'hidden' }}
+                      gap={2}
+                      align='baseline'
+                    >
+                      {paymentSource.paymentMethod === 'card' && (
+                        <Icon
+                          icon={CreditCard}
+                          sx={{ alignSelf: 'center' }}
+                        />
+                      )}
+                      <Text
+                        sx={t => ({ color: t.colors.$colorText, textTransform: 'capitalize' })}
+                        truncate
+                      >
+                        {paymentSource.paymentMethod === 'card' ? paymentSource.cardType : paymentSource.paymentMethod}
+                      </Text>
+                      <Text
+                        sx={t => ({ color: t.colors.$colorTextSecondary })}
+                        variant='caption'
+                        truncate
+                      >
+                        {paymentSource.paymentMethod === 'card' ? `⋯ ${paymentSource.last4}` : '-'}
+                      </Text>
+                      {paymentSource.isDefault && <Badge localizationKey={localizationKeys('badge__default')} />}
+                      {paymentSource.status === 'expired' && (
+                        <Badge
+                          colorScheme='danger'
+                          localizationKey={localizationKeys('badge__expired')}
+                        />
+                      )}
+                    </Flex>
+                    <PaymentSourceMenu paymentSource={paymentSource} />
+                  </ProfileSection.Item>
+
+                  <Action.Open value={`remove-${paymentSource.id}`}>
+                    <Action.Card variant='destructive'>
+                      <RemoveScreen
+                        paymentSource={paymentSource}
+                        revalidate={revalidate}
+                      />
```

Comment:
Seeing multiple spinners when deleting a payment source. I think we should only see it on the delete button and not over the content.

![Screenshot 2025-03-31 at 2 05 28 PM](https://github.com/user-attachments/assets/7610f682-2da5-4d5e-a0a2-315f6c9ebc07)